Add support to custom gitignore files

filipekiss opened the pull request:
This adds two new commands: `joe add` and `joe remove`. This way you can
add your own custom gitignore files to joe.

In case of adding a custom file already available in the default list,
the custom one will be used. This should work fine in Windows too, but I
wasn't able to test it. Any feedback is apreciated.

A new dependency was added: termcolor, for colored output.

Address karan/joe#9

filipekiss committed:
Add support to custom gitignore files.

This adds two new commands: `joe add` and `joe remove`. This way you can
add your own custom gitignore files to joe.

In case of adding a custom file already available in the default list,
the custom one will be used. This should work fine in Windows too, but I
wasn't able to test it. Any feedback is apreciated.

A new dependency was added: termcolor, for colored output.

The README file still needs to be updated.

filipekiss committed:
Fix problem with the Global ignores

The previous commit introduced a bug where the ignores located in the
Global folder would not work (for example, `osx`). This fix this
problem.

README still not updated.

filipekiss committed:
Updated the README to include the new commands.

This will update the README with the USAGE for `add` and `remove`.

karan commented:
Ok this will take a while to review and merge. Please make sure to pull recent changes and resolve the merge conflicts. Thanks for helping out @filipekiss

filipekiss committed:
Merge branch 'master' into feature/custom-ignores

Conflicts:
	joe/joe.py

filipekiss commented:
@karan just merged the master and pushed the changes and teste on my machine. Everything should be ok right now.

filipekiss commented:
I'm closing this for now. Just found a problem with it in Linux and filenames. I'll correct this and re-open the PR when fixed.

filipekiss committed:
Change the indexes for the files

Since now we are using colored displays, we need an index to keep the
filename and the cases since in a Case-Sensitive system this would cause
us trouble. I've tested this in Mac OS X and Linux, the first one case
insensite and the second case sensitive. It's working as expected.

filipekiss commented:
Found out the issue wasn't with Linux itself and actually with Case-Sensitve systems. Since my list was indexing things using only the lowercase name, I changed the way we use the indexes so we still maintain the filename.

filipekiss committed:
Update the version to the 0.0.6 in the setup script.

filipekiss committed:
Lowercase the name when looking for the index

karan commented:
Can you revert the commit that bumps the version? I'd like to bundle up a few PR's and then bump the version. :)

Also, tons of merge conflicts. :cry:

lorepozo commented on joe/joe.py:
```diff
@@ -24,15 +26,37 @@
+        'Linux' : os.path.join(_HOME, '.joe'),
+        'Windows' : os.path.join(_HOME, '.joe')
+    }
+    path = paths[PLATFORM] or None
```
I think what you're looking for here is `path = paths.get(PLATFORM)` to prevent a `KeyError`.

However, I don't think this is necessary. What's wrong with just `path = os.path.join(_HOME, '.joe')`? If I ran this in a Java Virtual Machine, or some machine that may return something not in `paths`, it would be marked as unsupported when that command should always work fine (unless there's some _deep_ problem, in which case python would just raise an exception).

filipekiss commented:
@Karin no problem. I'll also try to merge the master to prevent conflicts. :)

filipekiss commented:
As the pull-request itself is 5 years old and the project hasn't received a commit in almost 3 years, I think it's safe to close this.
